react/features/video-status-label/components/VideoStatusLabel.js

```diff
@@ -24,25 +24,112 @@ export class VideoStatusLabel extends Component {
         _audioOnly: React.PropTypes.bool,
 
         /**
+         * Whether or not a connection to a conference has been established.
+         */
+        _conferenceStarted: React.PropTypes.bool,
+
+        /**
          * Whether or not a high-definition large video is displayed.
          */
-        _largeVideoHD: React.PropTypes.bool
+        _largeVideoHD: React.PropTypes.bool,
+
+        /**
+         * Invoked to request toggling of audio only mode.
+         */
+        dispatch: React.PropTypes.func,
+
+        /**
+         * Invoked to obtain translated strings.
+         */
+        t: React.PropTypes.func
+    }
+
+    /**
+     * Initializes a new {@code VideoStatusLabel} instance.
+     *
+     * @param {Object} props - The read-only React Component props with which
+     * the new instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        // Bind event handler so it is only bound once for every instance.
+        this._toggleAudioOnly = this._toggleAudioOnly.bind(this);
     }
 
     /**
      * Implements React's {@link Component#render()}.
      *
      * @inheritdoc
-     * @returns {ReactElement|null}
+     * @returns {ReactElement}
      */
     render() {
-        if (this.props._audioOnly) {
-            return <AudioOnlyLabel />;
-        } else if (this.props._largeVideoHD) {
-            return <HDVideoLabel />;
+        const { _audioOnly, _conferenceStarted, _largeVideoHD, t } = this.props;
+
+        if (!_conferenceStarted || _largeVideoHD === undefined) {
+            return null;
+        }
+
+        let displayedLabel;
+
+        if (_audioOnly) {
+            displayedLabel = <i className = 'icon-visibility-off' />;
+        } else {
+            displayedLabel = _largeVideoHD
+                ? t('videoStatus.hd') : t('videoStatus.sd');
         }
 
-        return null;
+        return (
+            <div
+                className = 'video-state-indicator moveToCorner'
+                id = 'videoResolutionLabel' >
+                { displayedLabel }
+                { this._renderVideonMenu() }
+            </div>
+        );
+    }
+
+    /**
+     * Renders a dropdown menu for changing video modes.
+     *
+     * @private
+     * @returns {ReactElement}
+     */
+    _renderVideonMenu() {
+        const { _audioOnly, t } = this.props;
+
+        return (
+            <div className = 'video-state-indicator-menu'>
+                <div className = 'video-state-indicator-menu-options'>
+                    <div
+                        className = { _audioOnly && 'active' }
+                        { ...(!_audioOnly
```

Comment:
I would do here `...(!_audioOnly ? { onClick: this._toggleAudioOnly } : undefined)`.

Or actually you can build an object before these lines with the attributes for the divs and do `{ ...attributes }`. Maybe this will better.